c10/xpu/test/impl/XPUStreamTest.cpp

```diff
@@ -0,0 +1,205 @@
+#include <gtest/gtest.h>
+
+#include <c10/util/Optional.h>
+#include <c10/util/irange.h>
+#include <c10/xpu/XPUStream.h>
+
+#include <thread>
+#include <unordered_set>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+#define ASSERT_NE_XPU(X, Y) \
+  {                         \
+    bool isNE = X == Y;     \
+    ASSERT_FALSE(isNE);     \
+  }
```

Comment:
If it uses Gtest, why not rely on `EXECPT_EQ` and `EXCEPT_NE`?

Reply:
According to https://google.github.io/googletest/reference/assertions.html, 
>>Upon failure, EXPECT_ macros generate nonfatal failures and allow the current function to continue running, while ASSERT_ macros generate fatal failures and abort the current function.

Here, we need an assertation if failures are raised.

Reply:
Capture your point.  Change to `EXPECT_EQ` and `EXPECT_NE`